### PR TITLE
Simulator time/score and fonts

### PR DESCRIPTION
**Pull request by ericstraily**
Simulator Time and Score positions and colors, and adjusted the fonts a bit. Still need to add penalties to the simulator

### CODE REVIEW

File: uwh-refbox/src/main.rs [hunk 1]
```diff
@@ -140,53 +140,259 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
...
+                                .stroke(Some(game_color))
+                                .translate(Point::new(72, 33)),
+                        );
+                        display.draw(
+                            Font8x15::render_str("HALF")
+                                .stroke(Some(game_color))
+                                .translate(Point::new(68, 48)),
+                        );
+                    }
+                    _ => display.draw(
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: What about other game periods? We can have ref timeouts during any period...

File: uwh-refbox/src/main.rs [hunk 2]
```diff
@@ -140,53 +140,259 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
...
+                    Font22x46::render_str(&secs_to_time_string(state.secs_in_period))
+                        .stroke(Some(game_color))
+                        .translate(Point::new(74, 18)),
+                );
+                match state.current_period {
+                    GamePeriod::BetweenGames => display.draw(
+                        Font8x15::render_str("NEXT GAME")
+                            .stroke(Some(game_color))
+                            .translate(Point::new(92, 2)),
+                    ),
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: This whole thing can be much shorter:
```rust
let (text, x, y) = match state.current_period {
    GamePeriod::BetweenGames => ("NEXT GAME", 92, 2),
    GamePeriod::FirstHalf => ("1ST HALF", 100, 2),
    ...
}
display.draw(
    Font8x15::render_str(text)
        .stroke(Some(game_color))
        .translate(Point::new(x, y)),
);
```

File: uwh-refbox/src/main.rs [hunk 3]
```diff
@@ -140,53 +140,259 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
...
+                    GamePeriod::SuddenDeath => display.draw(
+                        Font8x15::render_str("SUDDEN DEATH")
+                            .stroke(Some(game_color))
+                            .translate(Point::new(80, 2)),
+                    ),
+                    _ => display.draw(
+                        Font8x15::render_str("ERROR_1")
+                            .stroke(Some(game_color))
+                            .translate(Point::new(64, 2)),
+                    ),
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: I think we don't want a default case in this match. If you remove it, the compiler will check that you've covered all the cases, which we want here

File: uwh-refbox/src/main.rs [hunk 4]
```diff
@@ -140,53 +140,259 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
...
+                }
+            }
+            _ => {
+                display.draw(
+                    Font16x31::render_str(&secs_to_time_string(state.secs_in_period))
+                        .stroke(Some(game_color))
+                        .translate(Point::new(108, 33)),
+                );
+                match state.current_period {
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: This can be shortened similarly to the above. Esp necessary with the next comment

File: uwh-refbox/src/main.rs (hunk 1 above)
Comment (TristanDebrunner) [reply uwh-refbox/src/main.rs]: This is another case where I think we don't want a default

File: uwh-refbox/src/main.rs [hunk 5]
```diff
@@ -140,53 +140,259 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
...
+                                .stroke(Some(timeout_color))
+                                .translate(Point::new(65, 4)),
+                        );
+                        display.draw(
+                            Font11x25::render_str("SHOT")
+                                .stroke(Some(timeout_color))
+                                .translate(Point::new(149, 4)),
+                        );
+                    }
+                    _ => display.draw(
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: No default, just have the `None` case do nothing

File: uwh-refbox/src/main.rs [hunk 6]
```diff
@@ -140,53 +140,259 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
...
+                        Font8x15::render_str("ERROR_2")
+                            .stroke(Some(game_color))
+                            .translate(Point::new(64, 2)),
+                    ),
+                }
+                match state.timeout {
+                    TimeoutState::White(_) => {
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: Switch to `TimeoutState::White(secs) => {` ...

File: uwh-refbox/src/main.rs [hunk 7]
```diff
@@ -140,53 +140,259 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
...
+                                .stroke(Some(timeout_color))
+                                .translate(Point::new(76, 2)),
+                        );
+                        display.draw(
+                            Font8x15::render_str("TIMEOUT")
+                                .stroke(Some(timeout_color))
+                                .translate(Point::new(68, 17)),
+                        );
+                        display.draw(
+                            Font16x31::render_str(":59")
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: ... then use `Font16x31::render_str(&format!(":{}", secs))`

File: uwh-refbox/src/main.rs [hunk 8]
```diff
@@ -140,53 +140,259 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
...
-            Font22x46::render_str("12:34")
-                .stroke(Some(white))
-                .translate(Point::new(64, 31)),
-        );
-        display.draw(
-            Font32x64::render_str("12:34")
-                .stroke(Some(red))
-                .translate(Point::new(128, 0)),
-        );
+        // EVERYTHING TO BE DISPLAYED ON THE WHITE SCORE PANEL
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: Eventually this needs to handle b and w switching sides. Can you add an issue for this?

File: uwh-refbox/src/main.rs [hunk 9]
```diff
@@ -1497,9 +1703,19 @@ where
...
+
+/*
+
+fn time_panel_center(text: &str) -> std::string::String
+{
+    text_length = text.len();
+    format!("{:^1$}",text, 16 + text_length)
 }
 
+*/
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: remove

File: uwh-refbox/src/main.rs [hunk 10]
```diff
@@ -1497,9 +1703,19 @@ where
 {
     let min = secs / T::from(60u8);
     let sec = secs % T::from(60u8);
-    format!("{:2}:{:02}", min, sec)
+    format!("{:>2}:{:>02}", min, sec)
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: Just curious, why this change? Right justify should be the default anyway

File: uwh-refbox/src/main.rs (hunk 1 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: I'm not sure I understand what you mean. These lines correspond to a section that is just displaying the text for 1st half or 2nd half in the case of a timeout. Do you mean `_ =>` when you say default?

File: uwh-refbox/src/main.rs (hunk 2 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: ah, yes, that's much nicer

File: uwh-refbox/src/main.rs (hunk 3 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: oh so the compiler can recognize that all options are listed?

File: uwh-refbox/src/main.rs (hunk 4 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: ok

File: uwh-refbox/src/main.rs (hunk 5 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: ok

File: uwh-refbox/src/main.rs (hunk 6 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: is this true for all instances of `(_)`?

File: uwh-refbox/src/main.rs (hunk 7 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: ok

File: uwh-refbox/src/main.rs (hunk 8 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: I will think about that

File: uwh-refbox/src/main.rs (hunk 9 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: yeah, I was trying to make my own function, didn't finish it, will delete

File: uwh-refbox/src/main.rs (hunk 10 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: oh, I thought left justify was the default

File: uwh-refbox/src/main.rs (hunk 1 above)
Comment (TristanDebrunner) [reply uwh-refbox/src/main.rs]: yes, the `_` arm of the match is called the default

File: uwh-refbox/src/main.rs (hunk 3 above)
Comment (TristanDebrunner) [reply uwh-refbox/src/main.rs]: yup

File: uwh-refbox/src/main.rs (hunk 6 above)
Comment (TristanDebrunner) [reply uwh-refbox/src/main.rs]: corresponds with the change in the next comment

Comment (TristanDebrunner) [reply uwh-refbox/src/main.rs]: `_` means "I don't care what this value is" `secs` means "I want to store this value to use later, and call it `secs`"

File: uwh-refbox/src/main.rs (hunk 1 above)
Comment (estraily-navisjon) [reply uwh-refbox/src/main.rs]: Can you address the first part of my response please?

File: uwh-refbox/src/main.rs [hunk 11]
```diff
@@ -140,53 +142,201 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
...
+                display.draw(
+                    Font16x31::render_str(&secs_to_time_string(state.secs_in_period))
+                        .stroke(Some(game_color))
+                        .translate(Point::new(108, 33)),
+                );
+            }
+        };
+
+        let black_penalty_count = 1;
+        let white_penalty_count = 1;
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: Should probably mark these with a comment as being interim

File: uwh-refbox/src/main.rs [hunk 12]
```diff
@@ -140,53 +142,201 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
...
+                );
+
+                let (text1, x1, y1, text2, x2, y2) = match state.current_period {
+                    GamePeriod::FirstHalf | GamePeriod::OvertimeFirstHalf => {
+                        ("1ST", 72, 33, "HALF", 68, 48)
+                    }
+                    GamePeriod::SecondHalf | GamePeriod::OvertimeSecondHalf => {
+                        ("2ND", 72, 33, "HALF", 68, 48)
+                    }
+                    _ => ("PERIOD ERROR", 72, 33, "", 0, 0),
```
Comment (TristanDebrunner) [inline uwh-refbox/src/main.rs]: Still have the problem that we can have a timeout in periods other than the ones specified. SD definitely can have ref t/o and penalty shot, and I currently allow ref t/o in any period for usability

File: uwh-refbox/src/main.rs (hunk 8 above)
Comment (TristanDebrunner) [reply uwh-refbox/src/main.rs]: Can you add a github issue to track this?

Comment (ericstraily) [reply uwh-refbox/src/main.rs]: ok. done

File: uwh-refbox/src/main.rs (hunk 11 above)
Comment (ericstraily) [reply uwh-refbox/src/main.rs]: ok

File: uwh-refbox/src/main.rs (hunk 12 above)
Comment (ericstraily) [reply uwh-refbox/src/main.rs]: I guess I don't understand what you are saying. What is a potential situation that is prohibited (or allowed) that shouldn't be?

Comment (TristanDebrunner) [reply uwh-refbox/src/main.rs]: The problem is that there are legitimate situations where we will see "PERIOD ERROR". Examples include a ref t/o or penalty shot during sudden death. Or, a ref t/o during half time, pre-{SD, OT}, or any other period not called first half or second half

Comment (ericstraily) [reply uwh-refbox/src/main.rs]: I can see the first example happening and will address in a follow-up reply, but for the second the timeout options should be suppressed during all game states except during game play

Comment (ericstraily) [reply uwh-refbox/src/main.rs]: for the first scenario, can I just not put anything in after the '_=>' and the program can just move to the next section of code?

Comment (TristanDebrunner) [reply uwh-refbox/src/main.rs]: You have to put something in. All branches of the match have to return the same type. I think for now just handling the SD case should be fine. I'll modify the ref t/o stuff so that's impossible, and we just stop the clock in other periods.

Comment (ericstraily) [reply uwh-refbox/src/main.rs]: ok done

**Comment by TristanDebrunner** [comment]
bors retry
